cogs/poetry.py

```diff
@@ -0,0 +1,85 @@
+        """
+		Generate a poem from one of our prebuilt configurations using our (not) state of the poetry generator!
+		Either call "?poem" for a random configration, or call "?poem x" where x is one of the valid prebuilts.
+		Valid prebuilts are: haiku, limerick and alexandrine
+        """
+        configs: List[List[Tuple[str,
+                                 int]]] = [[("A", 5), ("B", 7), ("C", 5)],
+                                           [("A", 10), ("A", 10), ("B", 7),
+                                            ("B", 7), ("A", 10)],
+                                           [("A", 12), ("A", 12), ("A", 12),
+                                            ("A", 12)]]
+        if command is None:
+            await ctx.trigger_typing()
+            await ctx.send("\n".join(self.poem_machine.mk_poem(
+                choice(configs))))
+            return
+        allowed_configs: List[str] = ["haiku", "limerick", "alexandrine"]
+        if command not in allowed_configs:
+            await ctx.send(
+                f"invalid option, valid options are: {allowed_configs}")
+            return
+        await ctx.trigger_typing()
+        if command == "haiku":
+            await ctx.send("\n".join(self.poem_machine.mk_poem(configs[0])))
+        elif command == "limerick":
+            await ctx.send("\n".join(self.poem_machine.mk_poem(configs[1])))
+        elif command == "alexandrine":
+            await ctx.send("\n".join(self.poem_machine.mk_poem(configs[2])))
+
+    @commands.command(aliases=["poem_custom"])
+    async def poemc(self, ctx, *, config_str: str = None):
+        """
+		Generate a poem from a custom configuration using our (not) state of the poetry generator!
+		Configurations are ordered as so: each line config is split by a space, where each line config
+		is first a letter (either capitalized or lowered), representing the rhyme of the line
+		(differently cased letters are parsed as different rhymes), and the rest
+		of the config is an integer, representing an amount of syllables for that line.
+		Examples of valid poemc calls:
+		> ?poemc A5 B7 A5 <- generates a haiku
+		> ?poemc A10 A10 B7 B7 A10 <- generates a limerick
+		> ?poemc o7 o7 o7 o7 o7 <- generates comradery
+        """
+        try:
+            poem_conf: List[Tuple[str, int]] = parse_poem_config(config_str)
+        except ValueError:
+            await ctx.send("invalid config")
+        else:
+            await ctx.trigger_typing()
+            await ctx.send("\n".join(self.poem_machine.mk_poem(poem_conf)))
+
+    @commands.command(aliases=["random_gen"])
+    async def genr(self, ctx, *, text_len: int = None):
+        """
+		Generate a random text using our (not) state of the text generator (fun fact: it generates text in reverse)!
+		Takes one command, which must specify the length of the text to be generated.
+        """
+        if text_len is None:
+            await ctx.send("length of text to generate must be specified")
```

Comment:
early return preferable here for indentings sake

Reply:
yep, good point, added early returns